projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -148,6 +151,7 @@ public function add_upgrades_menu( $plan = null ) {
 
 		if ( ! $menu_exists ) {
 			// Remove the submenu auto-created by Core.
+			// We remove it instead of hiding it because WP-Admin will see it as the first submenu item and therefore it acts as the menu's default URL.
```

Comment:
This comment confuses me. We're actually hiding the menu, aren't we? Why the comment says "We remove it instead of hiding it"?

Reply:
Oh yeah, good catch. That comment should be removed. When I started working on this PR I  replaced the hide_submenu_page with `remove_submenu_page` and added that comment. 

However, I've found another way to keep this and re-added hide_* method but forgot to remove that comment. :)

Reply:
> I've found another way to keep this and re-added hide_* method

Should we do the same then in `add_jetpack_menu`?

https://github.com/Automattic/jetpack/blob/6d33f193152bcde58d68271908fabacbb571a634/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L423

Reply:
I think we should leave it removed for jetpack, I think. I know we had some troubles with that submenu on non-admin pages. Either way it's fine, I think. :)

Reply:
That's fine, I don't see why the behavior would be different, but happy to have a diverging logic if that's causing any trouble. It'd be great if we could add a comment clarifying it.